packages/aggregator/src/analytics/aggregateAnalytics/aggregations/getFinalValuePerPeriod.js

```diff
@@ -57,14 +57,13 @@ class FinalValueCache {
     });
 
     this.cache = cache;
-    if (periodTypes.length > 1) {
-      // If multiple period types exist, make data consistent by converting
-      // all periods to the coarsest type (e.g. for day and month mix, convert to month)
-      this.ensurePeriodConsistency(findCoarsestPeriodType(periodTypes));
-    }
+
+    // Convert all the periods to a uniform aggregation period type
+    // eg: FINAL_EACH_MONTH will convert all periods to monthly periods
+    this.convertCachePeriods(aggregationPeriod);
```

Comment:
This is where the period conversion happens. I've smoke test all the affected reports and preaggregators and they seem alright.